### PR TITLE
(maint) Added support for signing RPMs using gpg >= 2.1

### PR DESCRIPTION
**Pull request by isaac-hammes**
Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

File: lib/packaging/sign/rpm.rb [hunk 1]
```diff
@@ -20,12 +20,20 @@ def sign(rpm, sign_flags = nil)
...
+    end
+
     # Try this up to 5 times, to allow for incorrect passwords
     Pkg::Util::Execution.retry_on_fail(:times => 5) do
       # This definition of %__gpg_sign_cmd is the default on modern rpm. We
       # accept extra flags to override certain signing behavior for older
       # versions of rpm, e.g. specifying V3 signatures instead of V4.
-      Pkg::Util::Execution.capture3("#{rpm_command} #{gpg_check_command} --define '%_gpg_name #{Pkg::Util::Gpg.key}' --define '%__gpg_sign_cmd %{__gpg} gpg #{sign_flags} #{input_flag} --batch --no-verbose --no-armor --no-secmem-warning -u %{_gpg_name} -sbo %{__signature_filename} %{__plaintext_filename}' --addsign #{rpm}")
+
+      Pkg::Util::Execution.capture3("#{rpm_command} #{gpg_check_command} --define '%_gpg_name #{Pkg::Util::Gpg.key}' --define '#{gpg_sign_cmd} #{sign_flags} #{input_flag} --batch --no-verbose --no-armor --no-secmem-warning -u %{_gpg_name} -sbo %{__signature_filename} %{__plaintext_filename}' --addsign #{rpm}")
```
[the remark this answers is not in the document]
Comment (isaac-hammes) [reply lib/packaging/sign/rpm.rb:36]: 1. The macro is either %__gpg_sign_cmd or %__gpg depending on the version of gpg. Would it make sense to cut it off at %__gpg and have the rest of the command in gpg_sign_command?
2. I like the changes for readability. Thank you

File: lib/packaging/sign/rpm.rb [hunk 2]
```diff
@@ -20,12 +20,34 @@ def sign(rpm, sign_flags = nil)
...
 
+    # If gpg version is >=2.1, use the gpg1 binary to sign. Otherwise, use the standard sign command.
+    gpg_executable = if gpg_version_greater_than_21?
+                       "%__gpg /usr/bin/gpg1' --define '%__gpg_sign_cmd %{__gpg} gpg1"
+                     else
+                       '%__gpg_sign_cmd %{__gpg} gpg'
+                     end
+
+    # rubocop:disable Lint/NestedPercentLiteral
+    gpg_sign = %W[
```
Comment (e-gris) [inline lib/packaging/sign/rpm.rb:31]: Suggest calling this 'gpg_signing_macro'

**Review by e-gris — COMMENTED** [review]
This looks great. I have one last small comment.

File: lib/packaging/sign/rpm.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (isaac-hammes) [reply lib/packaging/sign/rpm.rb]: > On point 1, I see my error; the macro was established in the assignment of gpg_sign_command.
> 
> I think we should think through the variable names to make this twisty bit of code more clear.
> 
> Happy to help with that.

Definitely. If you want to meetup after standup, that would be great. I'm not really certain what names make the most sense.

**Review by e-gris — APPROVED** [review]

**Pull request merged**